edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentCommand.cs

```diff
@@ -67,21 +70,41 @@ public class EdgeDeploymentCommand : ICommand
             };
         }
 
-        async Task UpdateImagePullSecrets(Dictionary<string, ImagePullSecret> imagePullSecrets, CancellationToken token)
+        public async Task ExecuteAsync(CancellationToken token)
+        {
+            await this.ManageImagePullSecrets(token);
+            await this.PushEdgeDeployment(token);
+        }
+
+        async Task ManageImagePullSecrets(CancellationToken token)
+        {
+            // Modules may share an image pull secret, so only pick unique ones to add to the dictionary.
+            List<ImagePullSecret> secrets = this.modules
+                .Select(module => this.combinedConfigProvider.GetCombinedConfig(module, this.runtimeInfo))
```

Comment:
```suggestion
                .Select(module => this.dockerConfigProvider.GetCombinedConfig(module, this.runtimeInfo))
```

Reply:
I deny this type of changes because code has changed